### PR TITLE
feature() : add document option

### PR DESCRIPTION
**Pull request by wandroll**
...ADD PR DETAILS HERE...

<details>
<summary>PR Checklist</summary>
<br/>

__Please leave this checklist in your PR.__

- Issue being fixed is referenced.
- Source changes maintain:
  - Stated browser compatibility.
  - Stated React compatibility.
- Unit test coverage added/updated.
- E2E test coverage added/updated.
- Prop-types added/updated.
- Typings added/updated.
- README updated (API changes, instructions, etc.).
- Changes to dependencies explained.
- Changeset added (run `yarn changeset` locally to add one, and follow the prompts).
  - EXCEPTION: A Changeset is not required if the change does not affect any of the source files that produce the package bundle. For example, tooling changes, test updates, or a new dev-only dependency to run tests more efficiently should not have a Changeset since it will not affect package consumers.

</details>

### CODE REVIEW

File: src/focus-trap-react.js [hunk 1]
```diff
@@ -87,9 +87,10 @@ class FocusTrap extends React.Component {
 
   /** Update the previously focused element with the currently focused element. */
   updatePreviousElement() {
-    if (typeof document !== 'undefined') {
-      this.previouslyFocusedElement = document.activeElement;
+    if (typeof this.props.focusTrapOptions.document !== 'undefined') {
+      this.previouslyFocusedElement = this.props.focusTrapOptions.document.activeElement;
     }
```
Comment (stefcameron) [inline src/focus-trap-react.js:92]: Maybe this, to cover the case where the option isn't specified?
```suggestion
    const doc = this.props.focusTrapOptions.document || document;
    if (doc) {
      this.previouslyFocusedElement = doc.activeElement;
    }
```

**Comment by stefcameron** [comment]
@wandroll FYI, the `master` branch has been updated with the new `focus-trap` 6.7.0 dependency I published earlier today, which includes the new `document` option.

File: demo/js/demo-iframe.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+          >
+            <div className="trap is-active">
+              <p>
+                Here is a focus trap inside an iframe <a href="#">with</a>{' '}
+                <a href="#">some</a> <a href="#">focusable</a> parts.
+              </p>
+              <p>
+                <button
+                  onClick={unmountTrap}
+                  aria-describedby="defaults-heading"
```
Comment (stefcameron) [inline demo/js/demo-iframe.js:89]: This is probably not referencing the right heading element... Should be `demo-iframe-heading` (per my suggestion in my previous comment).

File: demo/index.html [hunk 3]
```diff
@@ -131,6 +131,19 @@ <h2 id="setReturnFocus-heading">demo setReturnFocus option applied</h2>
         View demo source <span aria-hidden="true">&gt;&gt;</span>
       </a>
     </p>
+    <h2 id="demoIframe-heading">demo Iframe with document option applied</h2>
```
Comment (stefcameron) [inline demo/index.html:134]: For consistency, can we change the heading ID to `demo-iframe-heading`?

File: demo/index.html [hunk 4]
```diff
@@ -131,6 +131,19 @@ <h2 id="setReturnFocus-heading">demo setReturnFocus option applied</h2>
...
+    <p>
+      When integrated in an iframe, you may specify the document ( of the said
```
Comment (stefcameron) [inline demo/index.html:136]: Let's remove that extra space immediately after the `(`

File: demo/js/demo-iframe.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+            </div>
+          </FocusTrap>
+        )}
+      </SandboxContextConsumer>
+    </Sandbox>
+  );
+  return (
+    <div>
+      <p>
+        <button onClick={mountTrap} aria-describedby="defaults-heading">
```
Comment (stefcameron) [inline demo/js/demo-iframe.js:103]: `defaults-heading` -> `demo-iframe-heading`

**Review by stefcameron — COMMENTED** [review]
Just a few small easy updates, and we're good to go!

**Review by stefcameron — APPROVED** [review]
Thanks for the fixes! Just need to rebase one last time. I'd do it myself, but I can't modify your fork even though I'm an admin on focus-trap. I think that's because your `iadvize` org is different somehow.

**Comment by wandroll** [comment]
> Thanks for the fixes! Just need to rebase one last time. I'd do it myself, but I can't modify your fork even though I'm an admin on focus-trap. I think that's because your iadvize org is different somehow.

yes, its unfortunately not very flexible.

**Comment by stefcameron** [comment]
@all-contributors add @wandroll for code, test

**Comment by stefcameron** [comment]
Published in focus-trap-react v8.8.0

**Pull request merged**